packages/patternfly-4/react-core/src/components/DataList/DataListItem.tsx

```diff
@@ -12,32 +14,77 @@ export interface DataListItemProps extends Omit<React.HTMLProps<HTMLLIElement>,
   className?: string;
   /** Adds accessible text to the DataList item */
   'aria-labelledby': string;
+  /** Unique id for the DataList item */
+  id?: string;
 }
 
 export interface DataListItemChildProps {
   /** Id for the row */
   rowid: string;
 }
 
-export const DataListItem: React.FunctionComponent<DataListItemProps> = ({
-  children,
-  className = '',
-  isExpanded = false,
-  'aria-labelledby': ariaLabelledBy,
-  ...props
-}: DataListItemProps) => (
-  <li
-    className={css(styles.dataListItem, isExpanded && styles.modifiers.expanded, className)}
-    aria-labelledby={ariaLabelledBy}
-    {...props}
-  >
-    {React.Children.map(
-      children,
-      child =>
-        React.isValidElement(child) &&
-        React.cloneElement(child as React.ReactElement<any>, {
-          rowid: ariaLabelledBy
-        })
-    )}
-  </li>
-);
+export class DataListItem extends React.Component<DataListItemProps> {
+
+  // @ts-ignore
```

Comment:
Why do we need this?

Reply:
typescript has some issues with pulling in context this way. For a handful of versions of React, pulling in a static context was a deprecated practice. It is now supported again, but the ts linter doesn't know that? All instances of a static context in the project need the ts-ignore flag. And its the only way to reference the context outside of a context.consumer in a render function.